src/services/identity/SitesService.ts

```diff
@@ -125,64 +227,24 @@ class SitesService {
       },
     })
 
-    if (!site) {
-      return {
-        staging: "",
-        prod: "",
-      }
-    }
-
-    const deployment = site?.deployment
-    const output: SiteUrls = {
-      staging: deployment?.stagingUrl ? deployment.stagingUrl : "",
-      prod: deployment?.productionUrl ? deployment.productionUrl : "",
+    // Note: site may be null if the site does not exist
+    let siteUrls: SiteUrls = {
+      staging: site?.deployment?.stagingUrl ? site.deployment.stagingUrl : "",
+      prod: site?.deployment?.productionUrl
+        ? site.deployment.productionUrl
+        : "",
     }
 
-    for (const urlType of urlTypes) {
-      // We only call ConfigYmlService only when necessary
-      if (!output[urlType] && Object.keys(configYmlData).length === 0) {
-        const {
-          content: configData,
-        }: { content: ConfigYmlData } = await this.configYmlService.read(
-          sessionData
-        )
-
-        configYmlData = configData
-      }
-
-      if (!output[urlType] && urlType in configYmlData) {
-        const configSiteUrl = configYmlData[urlType]
-
-        if (configSiteUrl) {
-          output[urlType] = configSiteUrl
-        }
-      }
-
-      // We only call GitHubService only when necessary
-      if (!output[urlType] && repoDescTokens.length == 0) {
-        const {
-          description,
-        }: { description: string } = await this.gitHubService.getRepoInfo(
-          sessionData
-        )
-
-        // Retrieve the url from the description
-        // repo descriptions have varying formats, so we look for the first link
-        repoDescTokens = description.replace("/;/g", " ").split(" ")
-      }
+    siteUrls = await this.insertUrlsFromConfigYml(siteUrls, sessionData)
+    siteUrls = await this.insertUrlsFromGitHubDescription(siteUrls, sessionData)
```

Comment:
should aim to merge - lodash has a method for this iirc

Reply:
Improved in 42e73df.